src/main/java/org/broadinstitute/hellbender/dev/tools/walkers/bqsr/DF_BaseRecalibrator.java

```diff
@@ -76,68 +84,134 @@
 
-    @Argument(doc = "Number of Dataflow workers", optional = true)
-    protected int numWorkers = 0;
+            // 2. set up computation
+            PCollection<RecalibrationTables> aggregated =
+                    BQSR_Dataflow.GetRecalibrationTables(header, reads, referencePath, BRAC, skipIntervals);
 
-    // ------------------------------------------
+            if (null == outputTablesPath) {
+                // we need those, so let's pick a location for them.
+                if (remote) {
+                    outputTablesPath = GcsPath.fromUri(stagingLocation).resolve("temp-ds-recaltables.sj").toString();
+                } else {
+                    File outputTables = File.createTempFile("temp-ds-recaltables", ".sj");
+                    outputTables.deleteOnExit();
+                    outputTablesPath = outputTables.getPath();
+                }
+            }
+            boolean isGCS = serializeSingleObject(aggregated, outputTablesPath);
+            if (remote && !isGCS) {
+                throw new GATKException("If running on the cloud, either leave outputTablesPath unset or point it to a GCS location.");
+            }
+        } catch (RuntimeException rx) {
+            throw rx;
+        } catch (Exception x) {
+            throw new RuntimeException(x);
+        }
+    }
 
-    // the inputs to BQSR
-    private PCollection<Read> reads;
-    private PCollection<SimpleInterval> skipIntervals;
-    private SAMFileHeader header;
+    protected void afterPipeline(Pipeline p) {
+
+        //  Get the table back and output it in text form to RAC.RECAL_TABLE.
+        // TODO: if running on the cloud and the output destination is on the cloud, then it's faster to have a worker do it directly, without the file roundtrip.
+        final InputStream tblIn;
+        if (BucketUtils.isCloudStorageUrl(outputTablesPath)) {
+            try {
+                tblIn = Channels.newInputStream(new GcsUtil.GcsUtilFactory().create(p.getOptions()).open(GcsPath.fromUri(outputTablesPath)));
+            } catch (IOException e) {
+                throw new GATKException("Unexpected: unable to read results file. (bug?)", e);
+            }
+        } else {
+            try {
+                tblIn = new FileInputStream(outputTablesPath);
+            } catch (FileNotFoundException x) {
+                throw new GATKException("Unexpected: intermediate results file is missing. (bug?)", x);
+            }
+        }
+        ObjectInputStream oin;
+        try {
+            oin = new ObjectInputStream(tblIn);
+            Object o = oin.readObject();
+            RecalibrationTables rt = (RecalibrationTables) o;
+            baseRecalibratorUprooted.onTraversalStart(null);
+            baseRecalibratorUprooted.saveReport(rt, baseRecalibratorUprooted.getRequestedCovariates());
+        } catch (Exception e) {
+            throw new GATKException("Unexpected: unable to read results file. (bug?)", e);
+        }
```

Comment:
You do not appear to be closing any of your input/output streams. Use try-with-resources to ensure that they are closed even in the event of an exception.